cli/src/maintenance.rs

```diff
@@ -403,6 +433,53 @@ impl SolidoState {
         Some((instruction, task))
     }
 
+    /// If there is a validator being deactivated, try to unstake its funds.
+    // TODO(#386): Solana could fail the transaction to unstake, in that case,
+    // it should be retried.
+    pub fn try_unstake_from_inactive(&self) -> Option<(Instruction, MaintenanceOutput)> {
+        for validator in &self.solido.validators.entries {
+            // Validator is inactive, try to unstake from it.
+            if !validator.entry.active {
+                let try_unstake_balance = validator.entry.effective_stake_balance();
+                let (validator_stake_account, _) = validator.find_stake_account_address(
+                    &self.solido_program_id,
+                    &self.solido_address,
+                    validator.entry.stake_seeds.begin,
+                );
+                let (validator_unstake_account, _) = validator.find_unstake_account_address(
+                    &self.solido_program_id,
+                    &self.solido_address,
+                    validator.entry.unstake_seeds.end,
+                );
+                let task = MaintenanceOutput::UnstakeFromInactive {
+                    validator_vote_account: validator.pubkey,
+                    stake: validator_stake_account,
+                    unstake: validator_unstake_account,
+                    stake_seed: validator.entry.stake_seeds.begin,
+                    unstake_seed: validator.entry.unstake_seeds.end,
+                    amount: try_unstake_balance,
+                };
+
+                return Some((
+                    lido::instruction::unstake(
+                        &self.solido_program_id,
+                        &lido::instruction::UnstakeAccountsMeta {
+                            lido: self.solido_address,
+                            maintainer: self.maintainer_address,
+                            validator_vote_account: validator.pubkey,
+                            source_stake_account: validator_stake_account,
+                            destination_stake_account: validator_unstake_account,
```

Comment:
Oh, should we name this field `destination_unstake_account` in `UnstakeAccountsMeta`?

Reply:
Yeah, I think it's better, thanks